book/getting-started/run-a-wallaroo-application-docker.md

```diff
@@ -0,0 +1,201 @@
+
+- `--name wally`: The name for the container. This flag is optional but makes it easier to reference the container when we open other Bash sessions in the container.
+
+## Terminal 2, Start the Metrics UI
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+This command will start a new Bash shell within the container, we enter via the `environment-setup.sh` script to ensure we're in our persistent Python Virtual Environment.
+
+To start the Metrics UI run:
+
+```bash
+metrics_reporter_ui start
+```
+
+You can verify it started up correctly by visiting [http://localhost:4000](http://localhost:4000).
+
+If you need to restart the UI, run:
+
+```bash
+metrics_reporter_ui restart
+```
+
+When it's time to stop the UI, run:
+
+```bash
+metrics_reporter_ui stop
+```
+
+If you need to start the UI after stopping it, run:
+
+```bash
+metrics_reporter_ui start
+```
+
+## Terminal 3, Run Giles Receiver
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+We'll use Giles Receiver to listen for data from our Wallaroo application.
+
+```bash
+receiver --listen 127.0.0.1:5555 --no-write --ponythreads=1 --ponynoblock
+```
+
+You should see the `Listening for data` that indicates that Giles receiver is running.
+
+## Terminal 4, Run the "Celsius to Fahrenheit" Application
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+First, we'll need to get to the python Celsius example directory with the following command:
+
+```bash
+cd /src/wallaroo/examples/python/celsius
+```
+
+Now that we are in the proper directory, and the metrics UI and a data receiver are up and running, we can run the application itself by executing the following command:
+
+```bash
+machida --application-module celsius --in 127.0.0.1:7000 \
+  --out 127.0.0.1:5555 --metrics 127.0.0.1:5001 --control 127.0.0.1:6000 \
+  --data 127.0.0.1:6001 --name worker-name --external 127.0.0.1:5050 \
+  --cluster-initializer --ponythreads=1 --ponynoblock
+```
+
+This tells the "Celsius to Fahrenheit" application that it should listen on port `7000` for incoming data, write outgoing data to port `5555`, and send metrics data to port `5001`.
+
+## Terminal 5
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+### Sending Data with Giles Sender
+
+We will be sending in 25,000,000 messages using a pre-generated data file. The data file will be repeatedly sent via Giles Sender until we reach 25,000,000 messages.
```

Comment:
"We will send 25,000,000 messages using an included data file, which will be repeated by Giles Sender in order to reach 25,000,000 messages."

Reply:
I don't think this is much of an improvement on the original. In the original the word "repeatedly" is used, which indicates that something is done over and over again; the new one says "which will be repeated", which indicates that an action is done again, not done again and again. The new one also introduces some confusion about what is being "repeated".

You could try something like this:

> We will use Giles Sender to send 25,000,000 messages to the application. The messages come from an included data file, where each line in the file is used as a message. The sender loops over the file until it has sent 25,000,000 messages.

Then again, maybe it is sufficient to say:

> We will use Giles Sender to send 25,000,000 messages using an included data file.

And maybe provide a link to the Giles Sender documentation that the user can follow if they'd like to learn more about the specifics of how Giles Sender behaves.